raiden-ts/src/transport/epics.ts

```diff
@@ -83,6 +95,94 @@ import { getPresences$, getRoom$, roomMatch, globalRoomNames } from './utils';
 const AVAILABLE = ['online', 'unavailable'];
 const userRe = /^@(0x[0-9a-f]{40})[.:]/i;
 
+/**
+ * Joins the global broadcast rooms and returns the room ids.
+ *
+ * @param config - The {@link RaidenConfig} provides the broadcast room aliases for pfs and discovery.
+ * @param server - The matrix server used e.g., https://transport04.raiden.network is part of the alias.
+ * @param matrix - The {@link MatrixClient} instance used to create the filter.
+ * @returns Observable of the list of room ids for the the broadcast rooms.
+ */
+function joinGlobalRooms(
+  config: RaidenConfig,
+  server: string,
+  matrix: MatrixClient,
+): Observable<string[]> {
+  const serverName = getServerName(server);
+  return from(globalRoomNames(config)).pipe(
+    map(globalRoom => `#${globalRoom}:${serverName}`),
+    mergeMap(alias => matrix.joinRoom(alias)),
+    map(room => room.roomId),
+    toArray(), // wait all joinRoom promises to complete
+  );
+}
+
+/**
+ * Creates and returns a matrix filter. The filter reduces the size of the initial sync by
+ * filtering out broadcast rooms, emphemeral messages like receipts etc.
+ *
+ * @param matrix - The {@link MatrixClient} instance used to create the filter.
+ * @param roomIds - The ids of the rooms to filter out during sync.
+ * @returns Observable of the {@link Filter} that was created.
+ */
+function createFilter(matrix: MatrixClient, roomIds: string[]): Observable<Filter> {
+  return of(roomIds).pipe(
```

Comment:
All of this `of + map + map` is just composing a single object, it'd be clearer to have it defined directly (or in variables, if you wanna make steps explicit). The `mergeMap + from` can then be replaced by a `defer`. This makes all these operators a single `defer`, which is more concise.

Reply:
Splitting on different operators/*Maps is useful only for async chains